### PR TITLE
Ian development team comprehensive requirement

### PR DESCRIPTION
**Pull request by ianhomew**
- fixed problems in excel
- fixed some bugs
- create a new view to fast create landlord, landlord contract and relations.
- support multiupload with different input name.
- change validations to dynamic and set a global validator error messages.

### CODE REVIEW

File: config/whitelist.php [hunk 1]
```diff
@@ -13,7 +13,7 @@
         'id', 'name', 'certificate_number'
     ],
     'landlord_contracts' => [
-        'id','commission_start_date'
+        'id','commission_start_date', 'building_id'
```
Comment (nick66551) [inline config/whitelist.php]: Why do you want to display building_id here?

File: app/Http/Controllers/LandlordContractController.php [hunk 2]
```diff
@@ -74,16 +84,21 @@ public function store(Request $request)
...
             'agency_service_fee' => 'required',
             'rent_collection_frequency' => 'required|max:255',
             'rent_collection_time' => 'required|integer|digits_between:1,11',
             'rent_adjusted_date' => 'required|date',
-            'adjust_ratio' => 'required|numeric|between:0,99.99',
+            'adjust_ratio' => 'required|numeric|min:0',
             'deposit_month_count' => 'required|integer|digits_between:1,11',
             'is_collected_by_third_party' => 'required|boolean',
-            'is_notarized' => 'required|boolean'
+            'is_notarized' => 'required',
```
Comment (nick66551) [inline app/Http/Controllers/LandlordContractController.php]: If you create an enum rule, would you want to set this validation like `gender_limit`?

File: resources/views/landlord_contracts/form.blade.php [hunk 3]
```diff
@@ -275,17 +306,36 @@ class="form-control form-control-sm"
         }
     });
 </script>
+    <script id="event_in_form">
+        const $charter_fee = $('input[name=charter_fee]');
+
+        $('input[name=adjust_ratio]').on('change', function(){
```
Comment (nick66551) [inline resources/views/landlord_contracts/form.blade.php]: The function you should adjust is 
https://github.com/cchouse-erp/cchouse/blob/develop/app/Services/ScheduleService.php#L78

File: app/Http/Controllers/LandlordFastController.php [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+use App\ContactInfo;
+use App\Landlord;
+use App\LandlordAgent;
+use App\LandlordContract;
+use App\Responser\FormDataResponser;
+use App\Traits\Controllers\HandleDocumentsUpload;
+use Illuminate\Http\Request;
+use Illuminate\Validation\Rule;
+
+class LandlordFastController extends Controller
```
Comment (nick66551) [inline app/Http/Controllers/LandlordFastController.php]: Could you change this class as a service?
It's would be better if you make this as a Service like `RoomService`

File: routes/web.php [hunk 5]
```diff
@@ -28,6 +28,7 @@
             Route::resource('keys', 'KeyController');
             Route::resource('keyRequests', 'KeyRequestController');
             Route::resource('landlords', 'LandlordController');
+            Route::resource('landlordFast', 'LandlordFastController');
```
Comment (nick66551) [inline routes/web.php]: If you make this function as a Service, it would be better to remove this route resource

**Comment by nick66551** [comment]
Have you forgotten to do these? 
`房東合約管理` - `新增` : `6. 承租方式新增 "包租(業主)"，並確認有用到此欄位判斷的相關程式有做好對應的 else 機制`
`租約迄到期通知查看` : `要顯示: 相關的 Building 物件編號、完整地址、房東姓名、到期日`

File: config/whitelist.php (hunk 1 above)
Comment (ianhomew) [reply config/whitelist.php]: The building_id field is for sql join.

File: resources/views/landlord_contracts/form.blade.php (hunk 3 above)
Comment (ianhomew) [reply resources/views/landlord_contracts/form.blade.php]: Modified

File: app/Http/Controllers/LandlordContractController.php (hunk 2 above)
Comment (ianhomew) [reply app/Http/Controllers/LandlordContractController.php]: modified

File: app/Http/Controllers/LandlordFastController.php (hunk 4 above)
Comment (ianhomew) [reply app/Http/Controllers/LandlordFastController.php]: modified

File: routes/web.php (hunk 5 above)
Comment (ianhomew) [reply routes/web.php]: removed

**Pull request merged**